pom.xml

```diff
@@ -874,13 +874,13 @@
                 <groupId>org.jetbrains.kotlin</groupId>
                 <artifactId>kotlin-stdlib</artifactId>
                 <version>${kotlin.version}</version>
-                <scope>test</scope>
+                <scope>runtime</scope>
```

Comment:
The `scope` of these dependencies had to be changed so that places where they are transitive dependencies (e.g. through a `pmd-apex` dependency) see them as `runtime` rather than `test` (despite `pmd-apex/pom.xml` setting them to `compile`?). Please let me know if there's a better way to do this.

Reply:
> That would mean, we would miss the kotlin dependencies in `pmd-dist`, when we build the complete package... Did you see this problem there or somewhere else?

Yes, `pmd-dist` is where I first experienced this issue. The `pmd-apex` module would build correctly due to the Kotlin dependencies having `compile` scope in `pmd-apex/pom.xml`, but `pmd-dist` would incorrectly use the `test` scope from `pmd/pom.xml` when importing `pmd-apex`'s dependencies.

> this is in dependencyManagement - does the scope here actually have some effect? I thought it more as a recommendation - and you need to select the actual scope, when really declaring the dependency.

Dependencies in the child would inherit the `scope` defined in the parent's `dependencyManagement` if they're not explicitly specified. For example, if `pmd-apex/pom.xml` were
```xml
<dependency>
    <groupId>org.jetbrains.kotlin</groupId>
    <artifactId>kotlin-stdlib</artifactId>
</dependency>
<dependency>
    <groupId>org.jetbrains.kotlin</groupId>
    <artifactId>kotlin-stdlib-jdk8</artifactId>
</dependency>
```
, they would use `runtime` scope from `pmd/pom.xml`.

---

I cloned this branch again and I am actually no longer having this issue, even with `test` scope in `pmd/pom.xml`. I'm not sure why the results are different using the same inputs, but maybe this particular change (`test` -> `runtime`) can be reverted.

Reply:
> Dependencies in the child would inherit the scope defined in the parent's dependencyManagement if they're not explicitly specified.

I wasn't aware of this behavior.

I think in summary, it is more correct to declare kotlin-stdlib as runtime - since it really needs to be there at runtime for PMD (Apex, at least). I've seen, all other modules declare it explicitly as test dependency, so it should be ok.